### DISCUSSION TITLE
Adding a new index to existing table - will this skip indexing of current rows

### DISCUSSION THREAD

**Discussion by Shpadoinkle** [tywalch/electrodb#367] [Q&A]
Just a question regarding adding a new index and with a new gsi pk and sk to an existing table.   with composite values pointed at existing fields.

Does this apply a retroactive application of these values to the existing rows?   or will they be lost/excludes from the new queries on this index?

**Comment by raryanpur** [comment]
@Shpadoinkle it's automatic (DynamoDB handles this). See [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.OnlineOps.html#GSI.OnlineOps.Creating), `Phases of index creation -> Backfilling`.
